### PR TITLE
Log a proper error message when a valueset entry in a valuesets bundle doesn't contain version element

### PR DESCRIPTION
**Pull request by sanumula**
### Summary
Currently, when a valueset entry in a valuesets bundle doesn't contain version element a NullPointerException is thrown. The error message does not provide any details about which valueset entry has the missing element. This pull request addresses this issue by providing a proper error message with url information to identity the valueset entry in the bundle.

### CODE REVIEW

File: bunsen-spark-stu3/src/test/java/com/cerner/bunsen/stu3/codes/ValueSetsTest.java [hunk 1]
```diff
@@ -365,4 +376,34 @@ public void testExpandValues() {
...
+        .extractEntry(spark, valueSetsRdd, "ValueSet");
+    ValueSets valueSets = ValueSets.getEmpty(spark).withValueSets(valueSetList);
+
+    BroadcastableValueSets.Builder builder = BroadcastableValueSets.newBuilder();
+
+    Map<String, String> valueSetsInfo = new HashMap<>();
+    valueSetsInfo.put("HEDIS2020_Independent_Laboratory", "com:motivemi:valueset:mmi-33475");
+
+    for (Entry entry : valueSetsInfo.entrySet()) {
+
```
Comment (karthisuku) [inline bunsen-spark-stu3/src/test/java/com/cerner/bunsen/stu3/codes/ValueSetsTest.java:397]: nit: extra line

File: bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java [hunk 2]
```diff
@@ -153,6 +153,11 @@ protected AbstractValueSets(SparkSession spark,
     return spark.createDataset(members.rdd(), URL_AND_VERSION_ENCODER)
         .collectAsList()
         .stream()
+        .peek(p -> {
+          if (p.getVersion() == null) {
+            throw new RuntimeException("Version is null for valueset with url " + p.getUrl());
```
Comment (karthisuku) [inline bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java:158]: can `getUrl()` be `null`, in that case, will be any other information which we should log part of the message.

**Review by karthisuku — COMMENTED** [review]
few comments.

File: bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java (hunk 2 above)
Comment (amarvakul) [reply bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java:158]: We can look for `name` or `title` but none of them or mandatory from FHIR specifications so cannot be guaranteed to be present. I think we can stick to URL as this is the unique key element that's used in retrieving the valueset info programmatically, in most of the use cases.

Comment (karthisuku) [reply bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java:158]: Thanks @amarvakul . But is `url` is a mandatory field, if it is `null`, then the message is basically of no use :-(

Comment (amarvakul) [reply bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java:158]: Right @karthisuku, `url` is also not a mandatory field and when it is also not valued, the error message may not be completely helpful, but in the context where the error is currently being thrown (in `BroadcastableValuesets`), would at least give a lead/pointer that version and url are both empty.

Comment (karthisuku) [reply bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java:158]: Just a thought, Can we log the entire object, which would help us to identify which object is failing for triage purpose.

Comment (sanumula) [reply bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java]: @karthisuku What object are you referring to? If its UrlAndVersion then it has only 2 fields { url, version} which are logged in the error message.

Comment (karthisuku) [reply bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java]: Discussed offline, At this point during `peek` it does not have any more information otherthan `url` `version`.  In future if additional details are available we will enhance the logger as well.

**Review by karthisuku — APPROVED** [review]
+1 LGTM. Thanks for the details.

**Review by amarvakul — APPROVED** [review]
+1 LGTM. Thanks for the changes.

**Pull request merged**